components/camel-kafka/src/test/java/org/apache/camel/component/kafka/KafkaProducerTest.java

```diff
@@ -302,6 +306,21 @@ public void processSendsMessageWithMessageKeyHeader() throws Exception {
         assertRecordMetadataExists();
     }
 
+    @Test
+    public void processSendsMessageWithMessageTimestampHeader() throws Exception {
```

Comment:
If possible, would be nice to add one more test to test nested `OVERRIDE_TIMESTAMP` in nested messages, it is related to the observation here: https://github.com/apache/camel/pull/5312/files#r609685918